internal/git/gitsync_processor.go

```diff
@@ -96,15 +116,177 @@ func watchRepo(ctx context.Context, repo *v1.RepositoryPath, _ /* namespace */ s
 		logger.Debug("no monitoring")
 
 	} else {
-		// TODO: monitoring with intervals
-		logger.Debug("monitoring with intervals")
+		ticker := time.NewTicker(timeInterval * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				changedFiles, err := CheckForRepoUpdates(r, repo, &processor.gitSync.Status, ctx)
+				if err != nil {
+					return err
+				}
+				// apply the resources which are updated
+				for _, v := range changedFiles.ModifiedFiles {
+					// split the string on the basis of "---"
+					individualResource := strings.Split(string(v), "---")
+					logger.Debug(individualResource)
+				}
+
+				// delete the resources which are removed from file
+
+				// create the new resource
+
+			case <-ctx.Done():
+				logger.Debug("Context canceled, stopping updates check")
+				return nil
+			}
+		}
 	}
 	return nil
 }
 
-func NewGitSyncProcessor(ctx context.Context, gitSync *v1.GitSync, k8client client.Client, clusterName string) (*GitSyncProcessor, error) {
+// this check for file changes in the repo by comparing the old commit  hash with new commit hash
+
+func CheckForRepoUpdates(r *git.Repository, repo *v1.RepositoryPath, status *v1.GitSyncStatus, ctx context.Context) (PatchFiles, error) {
+	var patchedFiles PatchFiles
 	logger := logging.FromContext(ctx)
+	if err := fetchUpdates(r); err != nil {
+		logger.Errorw("error checking for updates in the github repo", "err", err)
+		return patchedFiles, err
+	}
+	remoteRef, err := getLatestCommit(r, repo.TargetRevision)
+	if err != nil {
+		logger.Errorw("failed to get latest commits in the github repo", "err", err)
+		return patchedFiles, err
+	}
+	lastCommitStatus := status.CommitStatus[repo.Name]
```

Comment:
hmm, need to think about thread safety if multiple RepositoryPaths touch this map. Will give more thought.

Reply:
maybe @xdevxy can address that as part of her PR to update the CommitStatus in the actual GitSync Resource